plugins/star-rating/api/api.js

```diff
@@ -10,30 +10,373 @@ var plugin = {},
+            if (params.qstring.trigger_font_color) changes["trigger_font_color"] = params.qstring.trigger_font_color;
+            if (params.qstring.target_devices) changes["target_devices"] = params.qstring.target_devices;
+            if (params.qstring.target_page) changes["target_page"] = params.qstring.target_page;
+            if (params.qstring.target_pages) changes["target_pages"] = params.qstring.target_pages || [];
+            if (params.qstring.is_active) changes["is_active"] = params.qstring.is_active;
+            if (params.qstring.hide_sticker) changes["hide_sticker"] = params.qstring.hide_sticker;
+
+            common.db.collection(collectionName)
+                .findAndModify({_id:common.db.ObjectID(id)},{}, changes, function(err, widget) {
+                    if (!err) {
+                        common.returnMessage(params, 200, 'Widget updated');
+                        return true;
+                    } else {
+                        common.returnMessage(params, 500, err.message);
+                        return false;
+                    }
+                })
+        }, params);
+        return true;
+    }
+
+    var removeWidgetData = function(widgetId, app, callback) {
+        var collectionName = "feedback" + app;
+
+        common.db.collection(collectionName)
+            .remove({"widget_id":widgetId},
+            function(err) {
+                if (!err) callback(null);
+                else callback(err);
+            })
+    }
+
+    plugins.register("/i", function(ob){
         var params = ob.params;
-        var events = (params.qstring && params.qstring.events);
-        if (events && events.length && Array.isArray(events)) {
-            events.forEach(function (event) {
-                if (event.key === '[CLY]_star_rating') {
-                    event.segmentation['platform_version_rate'] =
-                        event.segmentation['platform'] + "**" +
-                            event.segmentation['app_version'] + "**" +
-                            event.segmentation['rating'] + "**";
+        if (params.qstring.events && params.qstring.events.length && Array.isArray(params.qstring.events)) {
+            params.qstring.events = params.qstring.events.filter(function(currEvent){
+                if (currEvent.key == "[CLY]_star_rating"){
+                    
+                    common.db.collection('apps').findOne({'key': params.qstring.app_key}, (err, app) => {
+                        if (!app) {
+                            if (plugins.getConfig("api").safe) {
+                                common.returnMessage(params, 400, 'App does not exist');
+                            }
+                            return false;
+                        }
+                        
+                        var collectionName = 'feedback' + app["_id"];
+                        
+                        common.db.collection(collectionName)
+                            .insert({
```

Comment:
Here we insert feedback without even checking if feedback was provided at all? I mean if fields like comment is not even provided or is empty, do we need to insert it at all?